ftw/file/tests/test_view_name_equals_scale.py

```diff
@@ -0,0 +1,50 @@
+from ftw.file.testing import FTW_FILE_FUNCTIONAL_TESTING
+from plone.app.testing import TEST_USER_ID
+from plone.app.testing import TEST_USER_NAME
+from plone.app.testing import TEST_USER_PASSWORD
+from Products.CMFCore.utils import getToolByName
+from plone.app.testing import setRoles
+from plone.testing.z2 import Browser
+from unittest2 import TestCase
+from zope.component import getGlobalSiteManager
+from zope.interface import Interface
+from Products.Archetypes.interfaces.base import IBaseObject
+from zope.publisher.interfaces.browser import IDefaultBrowserLayer
+from Products.Five import BrowserView
+from ftw.builder import create
+from ftw.builder import Builder
+from ftw.testbrowser import browsing
+
+class TestView(BrowserView):
+
+    def __call__(self):
+        return "Success"
+
+
+class TestViewnameEqualsScale(TestCase):
+
+    layer = FTW_FILE_FUNCTIONAL_TESTING
+
+    def setUp(self):
+        gsm = getGlobalSiteManager()
+        setRoles(self.layer['portal'], TEST_USER_ID, ['Manager'])
+
+        gsm.registerAdapter(TestView,
+                            (IBaseObject,IDefaultBrowserLayer),
```

Comment:
You could use `ftw.testing`s [Component Registry Isolation Layer](https://github.com/4teamwork/ftw.testing#component-registry-isolation-layer) instead, so would not have to take care of cleaning up.
You probably would have to use `self.portal.getSiteManager()` then ...